src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +53,990 @@ const ActorInit Dm_Char08_InitVars = {
+                this->unk_203 = 0x63;
+                this->eyeMode = EYEMODE_1;
+                this->unk_207 = 0;
+                this->unk_208 = 0;
+                this->dyna.actor.flags |= ACTOR_FLAG_1;
+                if (gSaveContext.save.entranceIndex == 0x6A80) {
+                    this->eyeMode = EYEMODE_0;
+                    this->actionFunc = func_80AAFAC4;
+                } else {
+                    this->actionFunc = func_80AAF8F4;
+                }
+            } else {
+                this->actionFunc = func_80AAF610;
+            }
+            break;
+        case SCENE_SEA:
+            this->unk_1FF = 2;
+            this->animIndex = 2;
+            this->unk_203 = 0x63;
+            this->eyeMode = EYEMODE_0;
+            this->unk_207 = 0;
+            this->unk_208 = 0;
+            this->dyna.actor.flags |= ACTOR_FLAG_1;
+            this->actionFunc = func_80AAFAE4;
+            this->unk_1F0 = 1.0f;
+            break;
+        case SCENE_KONPEKI_ENT:
+            this->unk_1FF = 2;
+            this->animIndex = 2;
+            this->unk_203 = 0x63;
+            this->eyeMode = EYEMODE_0;
+            this->unk_207 = 0;
+            this->unk_208 = 0;
+            this->actionFunc = func_80AAFE78;
+            this->unk_1F0 = 1.0f;
+            break;
+    }
+    DmChar08_SetAnimation(&this->skelAnime, &sLargeTurtleAnimationInfo[this->animIndex], 0);
+}
+
+void DmChar08_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DmChar08* this = THIS;
+
+    if (this->dynapolyInitialized) {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+s16 D_80AB1770 = 0; // Some kind of latch for playing a sound only once?
+
+void func_80AAF610(DmChar08* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Player* player2 = GET_PLAYER(globalCtx);
+
+    if ((player2->stateFlags2 & 0x8000000) &&
+        ((player2->actor.world.pos.x > -5780.0f) && (player2->actor.world.pos.x < -5385.0f) &&
+         (player2->actor.world.pos.z > 1120.0f) && (player2->actor.world.pos.z < 2100.0f))) {
+        if (D_80AB1770 == 0) {
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            D_80AB1770 = 1;
+        }
+    } else {
+        D_80AB1770 = 0;
+    }
+    if ((player->transformation == PLAYER_FORM_ZORA) && (globalCtx->msgCtx.ocarinaMode == 3) &&
+        (globalCtx->msgCtx.unk1202E == 2)) {
+        if ((player2->actor.world.pos.x > -5780.0f) && (player2->actor.world.pos.x < -5385.0f)) {
+            if ((player2->actor.world.pos.z > 1120.0f) && (player2->actor.world.pos.z < 2100.0f)) {
+                this->actionFunc = func_80AAF79C;
+            }
+        }
+    }
+}
+
+void func_80AAF79C(DmChar08* this, GlobalContext* globalCtx) {
+    s16 cs1 = this->dyna.actor.cutscene;
+    s16 cs = ActorCutscene_GetAdditionalCutscene(
+        ActorCutscene_GetAdditionalCutscene(ActorCutscene_GetAdditionalCutscene(cs1)));
```

Comment:
Would be better to distinguish these a bit more. Notably cs1 is the one that actually gets examined.

Also, does this have to all be done in one go like this?